common/models/change.js

```diff
@@ -254,23 +278,35 @@ module.exports = function(Change) {
 
   /**
    * Get a change's current revision based on current data.
+   * @param  {Object} instance Optional instance object to get the revision string for, if not provided it will be queried
    * @callback  {Function} callback
    * @param {Error} err
    * @param {String} rev The current revision
    */
 
-  Change.prototype.currentRevision = function(cb) {
+  Change.prototype.currentRevision = function(instance, cb) {
+    // if only one argument was provided, then set that as the callback
+    if (typeof instance === 'function') {
+      cb = instance;
+      instance = null;
+    }
+
     cb = cb || utils.createPromiseCallback();
-    var model = this.getModelCtor();
-    var id = this.getModelId();
-    model.findById(id, function(err, inst) {
-      if (err) return cb(err);
-      if (inst) {
-        cb(null, Change.revisionForInst(inst));
-      } else {
-        cb(null, null);
-      }
-    });
+
+    if (instance) {
+      cb(null, Change.revisionForInst(instance));
```

Comment:
This does not make sense to me. Just replace calls of `change.currentRevision(inst)` with `change.constructor.revisionForInst(inst)` and then you don't need to change this method.